src/mpags-cipher.cpp

```diff
@@ -87,70 +60,53 @@ int main(int argc, char* argv[])
     // Initialise variables
     char inputChar{'x'};
     std::string inputText;
+    std::string outputText;
+    std::ifstream in_file;
+    std::ofstream out_file;
 
     // Read in user input from stdin/file
     // Warn that input file option not yet implemented
     if (!inputFile.empty()) {
-        std::cerr << "[warning] input from file ('" << inputFile
-                  << "') not implemented yet, using stdin\n";
-    }
-
-    // loop over each character from user input
-    while (std::cin >> inputChar) {
-        // Uppercase alphabetic characters
-        if (std::isalpha(inputChar)) {
-            inputText += std::toupper(inputChar);
-            continue;
+        in_file.open(inputFile);
```

Comment:
You could remove line 64 and here instead do:
```suggestion
        std::ifstream in_file { inputFile };
```
That way you also don't need the explicit close on line 82 since the object will go out of scope once we leave the if block.